### PR TITLE
Add correct access control headers to the default api config

### PR DESCRIPTION
**Pull request by Dignifiedquire**
Nedded to fix https://github.com/ipfs/js-ipfs-api/issues/76 and for https://github.com/ipfs/js-ipfs-api/issues/128 for details why this is needed see: https://github.com/ipfs/js-ipfs-api/issues/76#issuecomment-158216527

--- 

Probably lots wrong with this, so please excuse any obvious mistakes

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: b1b6fe7d1a099e78a9c9837ee2fbb2b1297e4696
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by: .* <.*>**

[Guidelines](https://github.com/ipfs/community/blob/master/contribution-guidelines.md#commit-messages) and [a script](https://github.com/ipfs/community/blob/master/dev/tools/hooks/setup_commit_msg_hook.sh) are available to help. Your feedback on GitCop is welcome on [this issue](https://github.com/ipfs/infrastructure/issues/23).

---
This message was auto-generated by https://gitcop.com

**Comment by diasdavid** [comment]
LGTM, thank you for catching this one @Dignifiedquire :)

For more context: https://botbot.me/freenode/ipfs/2015-11-19/?msg=54525489&page=9

**Comment by whyrusleeping** [comment]
@Dignifiedquire you'll need to fix the type your using in the config init:
```
HTTPHeaders: map[string][]string{
```

**Comment by diasdavid** [comment]
@whyrusleeping I was looking at that and wondering if it would make more sense to have the config coming from the JSON blob, instead of .go generating the config (this way could be one of those global things that other implementations use)

**Comment by whyrusleeping** [comment]
@diasdavid i'm not sure what youre saying. The JSON blob gets loaded into this struct so its easier to work with in go. (types are nice, mkay)

**Comment by diasdavid** [comment]
@whyrusleeping If I'm reading it right, when `ipfs init` is done, the config json file is created and then loaded from there. But what if we shipped the default config in a json file with the code?

**Comment by Dignifiedquire** [comment]
Will fix the type in the morning..

**Comment by whyrusleeping** [comment]
@diasdavid, some of the config is generated dynamically to make changing certain values easier.

File: repo/config/init.go [hunk 1]
```diff
@@ -44,6 +44,12 @@ func Init(out io.Writer, nBitsForKeypair int) (*Config, error) {
 			API:     "/ip4/127.0.0.1/tcp/5001",
 			Gateway: "/ip4/127.0.0.1/tcp/8080",
 		},
+		API: API{
+			HTTPHeaders: HTTPHeaders{
+				"Access-Control-Allow-Headers":  "X-Stream-Output, X-Chunked-Output",
+				"Access-Control-Expose-Headers": "X-Stream-Output, X-Chunked-Output",
+			},
+		},
```
Comment (jbenet) [inline repo/config/init.go]: add them also to `Gateway`, because gateway also has an API. (we need to rename these... not sure on good names)

Comment (Dignifiedquire) [inline repo/config/init.go]: @jbenet sorry not sure where exactly, do you mean on l. 83?

**Comment by Dignifiedquire** [comment]
@whyrusleeping I fixed the tests, anything else that I should do?

**Comment by jbenet** [comment]
I meant add the same headers to the "Gateway" config key as well.

**Comment by Dignifiedquire** [comment]
@jbenet done (or so I think)

**Comment by whyrusleeping** [comment]
this LGTM, though it seems kinda hacky, is this a config option we would ever *want* to change? if not, maybe it should go into the gateway code, and be hardcoded in?

**Comment by Dignifiedquire** [comment]
@whyrusleeping tbh you know better than I, these headers are directly dependent on the headers you are using in the api, so as long as you don't change those you don't have to change these here.

**Comment by Dignifiedquire** [comment]
@jbenet @whyrusleeping we really need this in 0.3.10 this is causing issues for a lot of people trying to use js-ipfs-api. Do you need anything else from my side?

**Comment by jbenet** [comment]
well, i think if this is _needed_ it shouldn't be a config value-- +1 to hardcoding this into where we set the defaults.

**Pull request merged**